File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Post.fs
namespace Back.Types.DataBase

open System
open System.ComponentModel.DataAnnotations

type Post(senderId: int64, title: string, content: string) =
    class
        [<Key>]
        member val PostId = 0L with get

        member val SenderId = senderId with get
        member val Title = title with get
        member val Content = content with get
        member val SendTime = DateTime.Now with get
        member val Checked = false with get, set
        member val CheckSucess = false with get, set
    end

File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Post.cs
using System.ComponentModel.DataAnnotations;

namespace Back.Types.DataBase;

public class Post
{
    [Key]
    public long PostId { get; set; }

    public long SenderId { get; set; }

    public string Title { get; set; }

    public string Content { get; set; }

    public DateTime SendTime { get; set; }

    public bool Checked { get; set; }

    public bool CheckSuccess { get; set; }

    public bool Deleted { get; set; }

    public Post(long senderId, string title, string content)
    {
        SenderId = senderId;
        Title = title;
        Content = content;
        SendTime = DateTime.Now;
        Checked = false;
        CheckSuccess = false;
        Deleted = false;
    }

    public Post()
    {
    }
}
